test/test.py

```diff
@@ -383,11 +384,9 @@ def _assert_bundle(self, uuid, version, manifest, metadata_files,
         bundle = Bundle(uuid, version, manifest, metadata_files)
 
         # Every data file's manifest entry should be referenced by a metadata
-        # entity that describes the data file. id() is used to work around the
-        # fact that dict instances aren't hashable and to ensure that no
-        # redundant copies are made.
-        self.assertEqual(set(id(f.manifest_entry.json) for f in bundle.files.values()),
-                         set(id(me) for me in manifest if not me['indexed']))
+        # entity that describes the data file.
+        self.assertEqual(list(f.manifest_entry.json for f in bundle.files.values()),
```

Comment:
I need to be convinced that this is either necessary or an improvement.

Reply:
This is necessary when I copied the JSON, but since I'm just casting now, it's unnecessary.